target/linux/ar71xx/base-files/lib/ar71xx.sh

```diff
@@ -581,8 +581,18 @@ ar71xx_board_detect() {
 		name="cpe505n"
 		;;
 	*"CPE510/520")
-		name="cpe510"
-		tplink_pharos_board_detect "$(tplink_pharos_get_model_string | tr -d '\r')"
+                name="cpe510"
+                tplink_v2_check() {
```

Comment:
I do not really get the reason for this function inside the switch/case block; in addition, it is repeated identically in platform.sh. By the way, this is also identical to the tplink_pharos_v2_get_model_string() introduced for the CPE210v2. So you could just remove the function and use md_string="$(tplink_pharos_v2_get_model_string)" later.